docs/getting_started/java/first_program_in_java/index.md

```diff
@@ -1,373 +1,757 @@
+  If you change this to `false`, the "deposit" fails and throws an exception.
+* The `refund process mimics the `deposit` process.
 
-        // WorkflowServiceStubs is a gRPC stubs wrapper that talks to the local Docker instance of the Temporal server.
-        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
-        WorkflowClient client = WorkflowClient.newInstance(service);
-        // Worker factory is used to create Workers that poll specific Task Queues.
-        WorkerFactory factory = WorkerFactory.newInstance(client);
-        Worker worker = factory.newWorker(Shared.MONEY_TRANSFER_TASK_QUEUE);
-        // This Worker hosts both Workflow and Activity implementations.
-        // Workflows are stateful so a type is needed to create instances.
-        worker.registerWorkflowImplementationTypes(MoneyTransferWorkflowImpl.class);
-        // Activities are stateless and thread safe so a shared instance is used.
-        worker.registerActivitiesImplementations(new AccountActivityImpl());
-        // Start listening to the Task Queue.
-        factory.start();
-    }
-}
-```
+:::note Recovering a failed Workflow
+
+It is not part of this tutorial, but you can try setting both `activityShouldSucceed` Booleans to false to see a Workflow Execution fail.
+Temporal lets you recover the Workflow.
+Stop the Worker, revert the changes, and restart
+With the "fixed" Worker running, use the Temporal Web UI (you'll read about it shortly) to reset the failed Workflow Execution to a point in its Event History where it can pick up and complete its work.
+
+:::
+
+## Set the Retry Policy
+
+Temporal makes software durable and fault-tolerant by design.
+The Service can "retry" Activities, which are failable by nature.
+That is, it can wait a little time for an issue to resolve, and then try running the Activity again, repeating as needed.
+You customize how this happens in a retry policy.
+That policy, along with time-out settings, is defined in the following Workflow implementation:
+
+<!--SNIPSTART money-transfer-java-workflow-implementation {"selectedLines": ["12-37"]}-->
 <!--SNIPEND-->
 
-Task Queues are defined by a simple string name.
+This code includes several components:
+
+* **`RetryOptions`** define how to automatically re-attempt an Activity that's failed.
+  This includes backoff coefficients (`setBackoffCoefficient`) that treat service providers' endpoints with respect, so you don't slam them at once.
+  Exponential backoffs allow a service time to recover.
+  In this code, if an Activity fails, the Service waits a second, then two seconds, then four, etc.
+  It fails after five attempts (`setMaximumAttempts`).
+  It won't wait more than 20 seconds between attempts (`setMaximumInterval`).
+* **`ActivityOptions`** define fixed limits for how long an Activity can keep trying before the Service shuts down further attempts.
+  *Always* set a schedule-to-close timeout limit.
```

Comment:
We don't recommend this. We say you must set either a Start-To-Close timeout or a Schedule-To-Close timeout but Start-To-Close is preferred because it is easier to estimate the entire time an Activity may need and not the scheduling aspects of Temporal, which can vary.

Reply:
Tiho lectured me because you can end up with infinite loops. I'm going to pass this by before making changes.